ui/pages/add-collectible/add-collectible.js

```diff
@@ -46,8 +50,68 @@ export default function AddCollectible() {
   );
   const [tokenId, setTokenId] = useState('');
   const [disabled, setDisabled] = useState(true);
+  const [eventTracked, setEventTracked] = useState([
+    {
+      event: 'Token Added',
+      address: '',
+      tracked: false,
+    },
+  ]);
+
   const [collectibleAddFailed, setCollectibleAddFailed] = useState(false);
 
+  const trackEvent = useContext(MetaMetricsContext);
+  const collectibles = useSelector(getCollectibles);
+
+  useEffect(() => {
+    if (!tokenId || !address) {
+      return;
+    }
+
+    const newCollectible = collectibles.find(
+      (collectible) =>
+        collectible.address.toLowerCase() === address.toLowerCase() &&
+        collectible.tokenId === tokenId.toString(),
+    );
+
+    if (!newCollectible) {
+      return;
+    }
+
+    const alreadyTracked = eventTracked.find(
+      (et) =>
+        et.address.toLowerCase() === address.toLowerCase() &&
+        et.tracked === true,
+    );
+
+    if (alreadyTracked) {
+      return;
+    }
+
+    trackEvent({
+      event: 'Token Added',
+      category: 'Wallet',
+      sensitiveProperties: {
+        token_contract_address: newCollectible.address,
+        tokenId: newCollectible.tokenId,
+        name: newCollectible.name || null,
+        description: newCollectible.description || null,
+        source: 'detected',
```

Comment:
A couple of broader questions adding to @brad-decker's point.
1. This is not a "detected" add token event. This is a manually added token. 
and 2. I'm a bit confused about why this track event is being called in a `useEffect` on the page where the user is adding the collectible. Is the idea that the value of `collectibles` will change when the user submits the form to add the collectible and this effect will happen before the component dismounts from the history.push event called on line 132 (at the end of the submission handler)?

Reply:
For 2: Yes. I did test this. The useEffect is called before the component dismounts. 
For 1: Do we have a "detected" flow for NFTs?

Reply:
oh. interesting. I was also confused by the original request, "when auto detection occurs for NFTs".

The existing patterns for "Token Added" do not live in `useEffect` hooks. If we were following the pattern used for "Token Added", we could call `trackEvent` in the same method or callback that adds the token. In this case, could we call track in `handleAddCollectible`?  I'm wondering where you found this pattern original pattern with `useEffect`